build.js

```diff
@@ -616,461 +891,536 @@ function copyIcons(app, outputPath) {
+      orientation: orientation,
+      studioName: config.studioName,
+      gameHash: app.manifest.version,
+      sdkHash: config.sdkVersion,
+      androidHash: androidVersion,
+      minSdkVersion: config.argv['min-sdk-version'] || 19,
+      targetSdkVersion: config.argv['target-sdk-version'] || 27,
+      debuggable: config.debug ? 'true' : 'false',
+      otherApps: otherApps.join('|')
     });
-}
 
-function updateActivity(config, opts) {
-  var activityFile = path.join(opts.outputPath,
-      "src",
-      config.packageName.replace(/\./g, "/"),
+    var defaultManifest = path.join(projectPath+"/app/src/main", "AndroidManifest.xml");
+    var outputManifest =  defaultManifest;
+    injectAppLinks(app.manifest.android, app)
+
+      .then(function () {
+        return injectPluginXML(opts, app);
+      })
+      .then(function () {
+        return Object.keys(opts.moduleConfig);
+      })
+      .map(function (moduleName) {
+        var module = opts.moduleConfig[moduleName];
+        var config = module.config;
+        if (config.injectionXSL) {
+          var xslPath = path.join(module.path, 'android', config.injectionXSL);
+          return transformXSL(api, defaultManifest, outputManifest, xslPath, params, config);
+        }
+      }, {concurrency: 1}) // Run the plugin XSLT in series instead of parallel
+
+      .then(function() {
+        /** Before this copy to original file seed of mygame/manifest.json:
+         copy 1: "fullscreen" : true,       // do not use strings like "true"
+         copy 2:  "gameHash" : 82378912738917238,
+         Otherwise you will seize following error in console
+         [error]  settings for AndroidManifest: value for fullscreen is not a string
+         [error]  settings for AndroidManifest: value for gameHash is not a string
+         */
+
+        logger.log("Applying final XSL transformation");
+        var xmlPath = path.join(projectPath,"/app/src/main", "AndroidManifest.xml");
+        return transformXSL(api, xmlPath, xmlPath,
+          path.join(__dirname, "AndroidManifest.xsl"),
+          params, config);
+      });
+  }
+
+  function setGradleParameters(app) {
+    var gradleFile = path.join(projectPath,
+      "app", "build.gradle");
+
+    return fs.readFileAsync(gradleFile, 'utf-8')
+      .then(function (contents) {
+
+        const versionCode = app.manifest.android.versionCode ? app.manifest.android.versionCode : "1"
+        var versionName = app.manifest.version ? app.manifest.version : "1.0"
+
+        contents = contents
+          .replace(/versionCode 1/g, "versionCode "+versionCode)
+          .replace(/versionName "1.0"/g,"versionName  \""+versionName+"\"")
+          .replace(/GameNamePlaceholderRelease/g, app.manifest.title)
+          .replace(/GameNamePlaceholderDebug/g, app.manifest.title+" debug");
+        return fs.writeFileAsync(gradleFile, contents);
+      });
+  }
+
+  function updateActivity(app, config, opts) {
```

Comment:
only config is used. no need of app and opts

Reply:
Done, config is used.